modules/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,467 @@
+    if (DS18B20_Is(ROM)==DEVICE_NOT_DS18B20) {
+        return DEVICE_NOT_DS18B20;
+    }
+    
+    /* Reset line */
+    if (OneWire_Reset(OneWire) == ONEWIRE_FAILURE) return DS18B20_FAILURE;
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWire, ROM);
+    /* Read scratchpad command by onewire protocol */
+    OneWire_WriteByte(OneWire, ONEWIRE_CMD_RSCRATCHPAD);
+    
+    /* Ignore first 2 bytes */
+    OneWire_ReadByte(OneWire, &temp);
+    OneWire_ReadByte(OneWire, &temp);
+    
+    OneWire_ReadByte(OneWire, &trigger_register_hi);
+    OneWire_ReadByte(OneWire, &trigger_register_lo);
+    OneWire_ReadByte(OneWire, &conf_register);
+    
+    /*  
+    ------------------------------------------
+    Configuration Register Resolution Values
+    ------------------------------------------
+    R0  R1  RESOLUTION_BITS
+    0   0   9
+    0   1   10
+    1   0   11
+    1   1   12
+    ------------------------------------------
+    */  
+    switch (resolution) {
+        case DS18B20_RESOLUTION_9BITS: {
+            conf_register &= ~(1 << DS18B20_RESOLUTION_R1);
+            conf_register &= ~(1 << DS18B20_RESOLUTION_R0);
+            break;
+        } 
+        case DS18B20_RESOLUTION_10BITS: {
+            conf_register &= ~(1 << DS18B20_RESOLUTION_R1);
+            conf_register |= 1 << DS18B20_RESOLUTION_R0;
+            break;
+        } 
+        case DS18B20_RESOLUTION_11BITS: {
+            conf_register |= 1 << DS18B20_RESOLUTION_R1;
+            conf_register &= ~(1 << DS18B20_RESOLUTION_R0);
+            break;
+        } 
+        case DS18B20_RESOLUTION_12BITS: {
+            conf_register |= 1 << DS18B20_RESOLUTION_R1;
+            conf_register |= 1 << DS18B20_RESOLUTION_R0;
+            break;
+        }
+        default: {
+            return DS18B20_USAGE_ERROR;
+            break;
+        }
+    }
+
+    /* Reset line */
+    if (OneWire_Reset(OneWire) == ONEWIRE_FAILURE) return DS18B20_FAILURE;
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWire, ROM);
+    /* Write scratchpad command by onewire protocol, only trigger_register_hi, trigger_register_lo and conf_register register can be written */
+    OneWire_WriteByte(OneWire, ONEWIRE_CMD_WSCRATCHPAD);
+    
+    /* Write bytes */
+    OneWire_WriteByte(OneWire, trigger_register_hi);
+    OneWire_WriteByte(OneWire, trigger_register_lo);
+    OneWire_WriteByte(OneWire, conf_register);
+    
+    /* Reset line */
+    OneWire_Reset(OneWire);
+    /* Select ROM number */
+    OneWire_SelectWithPointer(OneWire, ROM);
+    /* Copy scratchpad to EEPROM of DS18B20 */
+    OneWire_WriteByte(OneWire, ONEWIRE_CMD_CPYSCRATCHPAD);
+    
+    return DS18B20_SUCCESS;
+}
+
+DS18B20_Status DS18B20_Is(uint8_t* ROM) {
+    if (!ROM) return DS18B20_USAGE_ERROR;
+    /* Checks if first byte is equal to DS18B20's family code */
+    if (*ROM != DS18B20_FAMILY_CODE) {
+        return DEVICE_NOT_DS18B20;

```

Comment:
`return DS18B20_INVALID_DEVICE` ??

Reply:
Sorry find replace missed the file. Should be fixed now